extmod/modssl_mbedtls.c

```diff
@@ -270,13 +272,30 @@ STATIC mp_obj_t ssl_context_make_new(const mp_obj_type_t *type_in, size_t n_args
 
     return MP_OBJ_FROM_PTR(self);
 }
+STATIC int ssl_sock_cert_verify(void *ptr, mbedtls_x509_crt *crt, int depth, uint32_t *flags) {
+    mp_obj_ssl_socket_t *o = ptr;
+    if (o->ssl_context->handler == mp_const_none) {
+        return 0;
+    }
+    return mp_obj_get_int(mp_call_function_2(o->ssl_context->handler, mp_obj_new_bytes(crt->raw.p, crt->raw.len), MP_OBJ_NEW_SMALL_INT(depth)));
+}
+
+STATIC void ssl_context_set_cert_callback(mp_obj_ssl_context_t *self, mp_obj_t handler_in) {
+    if (handler_in != mp_const_none && !mp_obj_is_callable(handler_in)) {
+        mp_raise_ValueError(MP_ERROR_TEXT("invalid handler"));
+    }
+    self->handler = handler_in;
+}
+
 
 STATIC void ssl_context_attr(mp_obj_t self_in, qstr attr, mp_obj_t *dest) {
     mp_obj_ssl_context_t *self = MP_OBJ_TO_PTR(self_in);
     if (dest[0] == MP_OBJ_NULL) {
         // Load attribute.
         if (attr == MP_QSTR_verify_mode) {
             dest[0] = MP_OBJ_NEW_SMALL_INT(self->authmode);
+        } else if (attr == MP_QSTR_cert_callback) {
```

Comment:
I suggest calling this attribute `verify_callback`.  It indicates that it's related to the existing `verify_mode` attribute, and matches the CPython WIP PR: https://github.com/python/cpython/pull/31391